src/Akka.Persistence.Sql.Linq2Db/Journal/DAO/BaseByteArrayJournalDao.cs

```diff
@@ -374,31 +374,75 @@ await db.GetTable<JournalMetaData>()
+                }
+                else
+                {
+                    return MaxSeqForPersistenceIdQueryableNativeModeMinId(db, persistenceId, minSequenceNumber);
+                }
             }
-
-            if (_journalConfig.DaoConfig.SqlCommonCompatibilityMode)
+            else
             {
-                var nextQuery = db.GetTable<JournalMetaData>()
+                if (_journalConfig.DaoConfig.SqlCommonCompatibilityMode)
+                {
+                    return MaxSeqForPersistenceIdQueryableCompatibilityMode(db, persistenceId);
+                }
+
+                return MaxSeqForPersistenceIdQueryableNativeMode(db, persistenceId);
+            }
+        }
+
+        private static IQueryable<long?> MaxSeqForPersistenceIdQueryableNativeMode(DataConnection db,
+            string persistenceId)
+        {
+            return db.GetTable<JournalRow>()
+                .Where(r => r.persistenceId == persistenceId)
+                .Select(r => (long?)r.sequenceNumber);
+        }
+
+        private static IQueryable<long?> MaxSeqForPersistenceIdQueryableNativeModeMinId(
+            DataConnection db, string persistenceId, long minSequenceNumber)
+        {
+            return db.GetTable<JournalRow>()
+                .Where(r =>
+                    r.persistenceId == persistenceId &&
+                    r.sequenceNumber > minSequenceNumber)
+                .Select(r => (long?)r.sequenceNumber);
+        }
+
+        private static IQueryable<long?> MaxSeqForPersistenceIdQueryableCompatibilityModeWithMinId(
+            DataConnection db, string persistenceId, long minSequenceNumber)
+        {
+            return db.GetTable<JournalRow>()
+                .Where(r =>
+                    r.persistenceId == persistenceId &&
+                    r.sequenceNumber > minSequenceNumber)
+                .Select(r =>
+                    LinqToDB.Sql.Ext.Max<long?>(r.sequenceNumber).ToValue()).Union(db
+                    .GetTable<JournalMetaData>()
                     .Where(r =>
                         r.SequenceNumber > minSequenceNumber &&
-                        r.PersistenceId == persistenceId);
-                queryable = queryable.Union(nextQuery.Select(metaDataSelector));
-            }
+                        r.PersistenceId == persistenceId).Select(r =>
+                        LinqToDB.Sql.Ext.Max<long?>(r.SequenceNumber).ToValue()));
+        }
 
-            return queryable.OrderByDescending(sequenceNumberSelector)
-                .Select(sequenceNumberSelector).Take(1);
+        private static IQueryable<long?> MaxSeqForPersistenceIdQueryableCompatibilityMode(
+            DataConnection db, string persistenceId)
+        {
+            return db.GetTable<JournalRow>()
+                .Where(r => r.persistenceId == persistenceId).Select(r =>
+                    LinqToDB.Sql.Ext.Max<long?>(r.sequenceNumber).ToValue()).Union(db
```

Comment:
So, there is 'technically' a way to call this version of `.Max()` via Extension method, but of course it collides with `IEnumerable<T>.Max()`. We are coercing the nullable long here so that we can `.GetValueOrDefault()` in the main call.

But basically, we wind up getting something like below. There -might- be a better way to write the query and avoid the parameter dupe.

```sqlite
SELECT
	Max([t3].[c1])
FROM
	(
		SELECT
			[t1].[c1]
		FROM
			(
				SELECT
					MAX([r].[sequence_number]) as [c1]
				FROM
					[journal] [r]
				WHERE
					[r].[persistence_id] = @persistenceId
			) [t1]
		UNION
		SELECT
			[t2].[c1]
		FROM
			(
				SELECT
					MAX([r_1].[sequenceNr]) as [c1]
				FROM
					[journal_metadata] [r_1]
				WHERE
					[r_1].[PersistenceId] = @persistenceId_1
			) [t2]
	) [t3]
```

Reply:
Looks just like what we have in Akka.Persistence.Sql.Common - I think it's fine. Thanks for spelling out the generated SQL.